client/src/main/java/org/evosuite/ga/metaheuristics/CellularGA.java

```diff
@@ -0,0 +1,369 @@
+				applyLocalSearch();
+				double bestFitnessAfterLocalSearch = getBestFitness();
+
+				if (getFitnessFunction().isMaximizationFunction())
+					assert(bestFitnessAfterLocalSearch >= (bestFitnessBeforeLocalSearch
+							- DELTA)) : "best fitness before applyLocalSearch() was: " + bestFitnessBeforeLocalSearch
+									+ ", now best fitness is " + bestFitnessAfterLocalSearch;
+				else
+					assert(bestFitnessAfterLocalSearch <= (bestFitnessBeforeLocalSearch
+							+ DELTA)) : "best fitness before applyLocalSearch() was: " + bestFitnessBeforeLocalSearch
+									+ ", now best fitness is " + bestFitnessAfterLocalSearch;
+			}
+
+			double newFitness = getBestFitness();
+
+			if (getFitnessFunction().isMaximizationFunction())
+				assert(newFitness >= (bestFitness - DELTA)) : "best fitness was: " + bestFitness
+						+ ", now best fitness is " + newFitness;
+			else
+				assert(newFitness <= (bestFitness + DELTA)) : "best fitness was: " + bestFitness
+						+ ", now best fitness is " + newFitness;
+			bestFitness = newFitness;
+
+			if (Double.compare(bestFitness, lastBestFitness) == 0) {
+				starvationCounter++;
+			} else {
+				logger.info("reset starvationCounter after " + starvationCounter + " iterations");
+				starvationCounter = 0;
+				lastBestFitness = bestFitness;
+
+			}
+
+			updateSecondaryCriterion(starvationCounter);
+
+			logger.info("Current iteration: " + currentIteration);
+			this.notifyIteration();
+
+			logger.info("Population size: " + population.size());
+			logger.info("Best individual has fitness: " + population.get(0).getFitness());
+			logger.info("Worst individual has fitness: " + population.get(population.size() - 1).getFitness());
+
+		}
+		// archive
+		TimeController.execute(this::updateBestIndividualFromArchive, "update from archive", 5_000);
+
+		notifySearchFinished();
+	}
+	
+	/**
+	 * Retrieve the fitness
+	 * @return fitness of an individual
+	 */
+	private double getBestFitness() {
+		T bestIndividual = getBestIndividual();
+		for (FitnessFunction<T> ff : fitnessFunctions) {
+			ff.getFitness(bestIndividual);
+		}
+		return bestIndividual.getFitness();
+	}
+	
+	/**
+	 * <p>
+	 * setReplacementFunction
+	 * </p>
+	 * 
+	 * @param replacement_function
+	 *            a {@link org.evosuite.ga.ReplacementFunction} object.
+	 */
+	public void setReplacementFunction(ReplacementFunction replacement_function) {
+		this.replacementFunction = replacement_function;
+	}
+
+	/**
+	 * <p>
+	 * getReplacementFunction
+	 * </p>
+	 * 
+	 * @return a {@link org.evosuite.ga.ReplacementFunction} object.
+	 */
+	public ReplacementFunction getReplacementFunction() {
+		return replacementFunction;
+	}
+	
+	/**
+	 * Determine the offspring that will be kept
+	 * @param offspring1 The first offspring
+	 * @param offspring2 The second offspring
+	 * @return a boolean
+	 */
+	@SuppressWarnings("deprecation")
+	protected boolean keepOffspring(Chromosome offspring1, Chromosome offspring2) {
```

Comment:
If we use the isBetterOrEqual method from GeneticAlgorithm class, we could remove this method.